File: src/core/audit/status.ts
import { readFile } from 'node:fs/promises';
import { resolve } from 'node:path';

const AUDIT_LOG_PATH = resolve('data', 'audit.log.jsonl');

export type AuditLogStatus = {
  lastEventTime?: string;
  entries: number;
};

export const auditLogStatus = async (): Promise<AuditLogStatus> => {
  try {
    const content = await readFile(AUDIT_LOG_PATH, 'utf8');
    const lines = content.split('\n').filter(Boolean);
    const last = lines.at(-1);
    if (!last) {
      return { entries: 0 };
    }

    const parsed = JSON.parse(last) as { timestamp?: string };
    return { entries: lines.length, lastEventTime: parsed.timestamp };
  } catch (error: unknown) {
    if ((error as NodeJS.ErrnoException).code === 'ENOENT') {
      return { entries: 0 };
    }
    throw error;
  }
};

File: tornado_ai/core/audit/status.py
"""Helpers for reading the JSONL audit log."""
from __future__ import annotations

import json
from dataclasses import dataclass
from pathlib import Path
from typing import Optional

AUDIT_LOG_PATH = Path("data") / "audit.log.jsonl"


@dataclass
class AuditLogStatus:
    entries: int
    last_event_time: Optional[str] = None


def audit_log_status() -> AuditLogStatus:
    try:
        content = AUDIT_LOG_PATH.read_text(encoding="utf-8")
    except FileNotFoundError:
        return AuditLogStatus(entries=0)
    lines = [line for line in content.splitlines() if line.strip()]
    if not lines:
        return AuditLogStatus(entries=0)
    last = json.loads(lines[-1])
    return AuditLogStatus(entries=len(lines), last_event_time=last.get("timestamp"))
